### PR TITLE
Add LIGO contracts

### PR DESCRIPTION
**Pull request by kkirka**
Currently, the repo only contains the compiled versions of the contracts. However, people want to review the source code of the contracts, and, eventually, maybe even verify the contracts with tools like etherscan (if such tools appear on Tezos). Adding the source code would increase the trust in the tool, and facilitate its promotion.

This PR:
- [x] adds tqtezos contracts as a submodule dependency
- [x] adds LIGO contracts for Mintable/Fixed Pausable/NotPausable scenarios
- [x] updates the compiled bytecode to the one produced by LIGO v0.17 (the latest LIGO by now)
- [x] adds `npm run compile-contracts` script to compile the contracts
- [x] fixes a small typo in contracts' names

See the corresponding commits for the details.

Resolves https://github.com/KStasi/fa2-deployer/issues/8

### CODE REVIEW

**Comment by KStasi** [comment]
Thank you very much for the commitment! It definitely had to be done.
 I need some time to fix Netlify builds and the PR will be merged.
 
 Now it can't clone the repo with the contracts:
 
 ```
 fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:tqtezos/smart-contracts.git' into submodule path '/opt/build/repo/src/contracts/tqtezos/smart-contracts' failed
Failed to clone 'src/contracts/tqtezos/smart-contracts'. Retry scheduled
Cloning into '/opt/build/repo/src/contracts/tqtezos/smart-contracts'...
Host key verification failed.
 ```

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "src/contracts/tqtezos/smart-contracts"]
+	path = src/contracts/tqtezos/smart-contracts
+	url = git@github.com:tqtezos/smart-contracts.git
```
Comment (KStasi) [inline .gitmodules:3]:
```suggestion
	url = https://github.com/tqtezos/smart-contracts.git
```

**Comment by kkirka** [comment]
It wants `typescript` now, and I couldn't find a straightforward way to overcome this :disappointed:
I can add a typescript dependency ofc but I want to know if there are other options. Maybe I could just copy the relevant contracts instead of making it a submodule? Upgrades may become a bit more painful but as I see the situation we have two chairs now, basically: either depend on typescript or get rid of the submodule. Which one you think is preferable? Or maybe there's some other option that I'm missing?

**Comment by clemsos** [comment]
hi @kkirka thanks for the work on the PR! @KStasi I am the person that asked about the contracts #8

I personally have only bad experiences using git submodules. Also considering this repo is not an actively maintained  project, seems fine to just check the ligo files in for people to look at.

Maybe a straightfoward way will be to fetch the contracts directly from the tqtezos repo using npm? I asked them, let's see what they say https://github.com/tqtezos/smart-contracts/issues/70

**Comment by KStasi** [comment]
I have added the contracts, updated Michelson, and the compilation instruction into the master branch. It's not that cool as submodules but it works. 
Hope now the project is more dev-friendly.

**Pull request closed without merging**